portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2085,6 +2087,16 @@ private void _addFileEntryResources(
 				dlFileEntry.getFileEntryId(),
 				serviceContext.getModelPermissions());
 		}
+
+		long inheritableParentFolderId =
+			DLPermissionPropagationUtil.getInheritableParentFolderId(
+				dlFileEntry);
+
+		if (FeatureFlagManagerUtil.isEnabled("LPS-87806") &&
+			(inheritableParentFolderId >= 0)) {
+
+			_inheritRolesPermissions(inheritableParentFolderId, dlFileEntry);
+		}
```

Comment:
It should be refactored as below instead of setting the inherited permissions after the default document permissions have been set
```
		long inheritableParentFolderId =
			DLPermissionPropagationUtil.getInheritableParentFolderId(
				dlFileEntry);

		if (FeatureFlagManagerUtil.isEnabled("LPS-87806") &&
			(inheritableParentFolderId >= 0)) {

			_inheritRolesPermissions(inheritableParentFolderId, dlFileEntry);
		} else {
		
			//Adding the default document permissions
		}
```

Reply:
We can't do that, because when uploading new file, default document permissions are added first, then common permissions are added.

Reply:
I see. Thanks.